### ISSUE TITLE
Decomposition of Non-H1 spaces doesn't work on geometries with repeated knots.

### ISSUE THREAD

**Issue by anlavandier** [pyccel/psydac#229]
I found a bug when discretizing VectorFemSpace on the geometry pipe.h5 when kind is not h1. It seems to be caused by the repeated knots.
Here's the code :
```
import os
from mpi4py import MPI
from sympde.topology import VectorFunctionSpace, Domain, ScalarFunctionSpace
from psydac.api.discretization import discretize

geometry = 'pipe.h5'
kind = 'hcurl'
space = VectorFunctionSpace


comm = MPI.COMM_WORLD
rank = comm.Get_rank()

filename = os.path.join('../mesh', geometry)

domain = Domain.from_file(filename=filename)
domain_h = discretize(domain, filename=filename, comm=comm)

dim = domain.dim

symbolic_space = space('V', domain, kind=kind)

mapping = list(domain_h.mappings.values())[0]
degree = mapping.space.degree

space_h = discretize(symbolic_space, domain_h, degree=degree)

print(rank, f"mapping local domain {mapping.space.local_domain}\n",
            f"space.spaces[0] local domain {space_h.spaces[0].local_domain}\n",
            f"space.spaces[1] local domain {space_h.spaces[1].local_domain}\n")
```
When called with `mpirun -n 2`, the domain decomposition is not what it should be :
```
1 mapping local domain ((15, 0), (31, 31))
 space.spaces[0] local domain ((8, 0), (31, 31))
 space.spaces[1] local domain ((7, 0), (31, 31))

0 mapping local domain ((0, 0), (14, 31))
 space.spaces[0] local domain ((0, 0), (7, 31))
 space.spaces[1] local domain ((0, 0), (7, 31))
 ```
This is done using `sympde==0.14.1` as it fails on newer versions.

**Comment by yguclu** [comment]
@saidctb Should this issue be fixed by PR #232?

**Comment by saidctb** [comment]
> @saidctb Should this issue be fixed by PR #232?

yes, it's fixed by the PR #232
